### DISCUSSION TITLE
Navigation entre menus

### DISCUSSION THREAD

**Discussion by RomainCuchet** [MorganKryze/ConsoleAppVisuals#138] [Ideas]
Bonjour,
Pour le projet de modélisation de base de données j'avais besoin de générer automatiquement des ScrollingMenu. Depuis un Scrolling menu je voulais pouvoir accéder à d'autres ScrollingMenus etc. De plus à tout moment je voulais pouvoir faire flèche gauche et retourner sur le menu précédent. J'ai Voulu faire une classe menu qui prenait en paramètre le menu précédent mais il y avait un blocage car je ne pouvais pas définir un ScrollingMenu en tant que champ non statique. Au final j'ai développé le truc moi même sans passer par ta librairie mais ce serait une feature super cool.

**Comment by RomainCuchet** [comment]
Les switch case sont aussi un peu encombrants. J'ai fait un système plutôt sympa pour s'en passer. Si ça t'intéresse je  t'envoie mon code

**Reply by MorganKryze** [reply]
Avec plaisir, je n'ai pas beaucoup exploré les alternative si ce n'est séries de if/else ou if sur des cas spécifiques. N'hésite pas à me contacter par mail au besoin.

**Comment by MorganKryze** [comment]
Bonjour Romain,
Tout d'abord merci pour l'intérêt que tu portes à la librairie, et d'avoir pris le temps d'expliquer ton problème !

Pour ce qui est de la navigation entre menus, elle est "abordée" dans les différents exemples mis à disposition dans la documentation ([voici un exemple](https://morgankryze.github.io/ConsoleAppVisuals/4-examples/int-password.html)). C'est donc déjà une feature prise en charge par défaut par la librairie. Mais de ce que je comprends cela va plus loin puisque tu parles de génération automatique. Je veux bien que tu développes dessus pour mieux comprendre ton cas d'usage.

Concernant le retour en arrière, il est généralement utilisé la touche [escape] que tu traites pour revenir au menu précédent (voir exemples dispo dans la documentation). Si cela ne te va pas et que  la question pour toi est le choix et l'utilité des touches rendues possibles à l'utilisateur, alors tu as deux choix si tu le souhaites : [créer ton propre élément](https://morgankryze.github.io/ConsoleAppVisuals/5-articles/create_element.html) (et le partager si tu le souhaite) ou créer une feature request pour qu'on puisse mettre ça en place à l'échelle de la librairie.

Encore merci pour ton retour !
